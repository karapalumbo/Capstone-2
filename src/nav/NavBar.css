.nav-container {
  border-bottom: 2px solid lightgray;
  padding: 0.5rem 1rem;
  justify-content: space-between !important;
  margin-bottom: 20px;
}

.nav-links {
  display: inline-flex;
  font-weight: 600;
}

a {
  color: black !important;
}

.selected {
  color: #805ad5 !important;
}

.brand {
  color: #553c9a !important;
}
