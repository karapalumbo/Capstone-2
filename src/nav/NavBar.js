import React, { useContext } from "react";
import UserContext from "../UserContext";
import { NavLink, Navbar, NavbarBrand } from "reactstrap";
import { Link } from "react-router-dom";
// import "./NavBar.css";

function NavBar({ logout }) {
  const { currentUser } = useContext(UserContext);

  return (
    <Navbar className="nav-container" color="light" expand="md">
      <NavbarBrand href="/">Petfinder</NavbarBrand>
      <div className="nav-links">
        {currentUser ? (
          <>
            <NavLink href="/companies">Pets</NavLink>
            <NavLink href="/jobs">Favorites</NavLink>
            <NavLink href="/profile">Profile</NavLink>
            <Link className="nav-link" to="/" onClick={logout}>
              Logout {currentUser.username}
            </Link>
          </>
        ) : (
          <>
            <NavLink href="/login">Login</NavLink>
            <NavLink href="/signup">Signup</NavLink>
          </>
        )}
      </div>
    </Navbar>
  );
}

export default NavBar;
