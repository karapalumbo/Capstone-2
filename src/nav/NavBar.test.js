import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NavBar from "./NavBar";
import UserTestProvider from "../testHelper";

it("matches snapshot when logged in", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserTestProvider>
        <NavBar />
      </UserTestProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserTestProvider currentUser={null}>
        <NavBar />
      </UserTestProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
