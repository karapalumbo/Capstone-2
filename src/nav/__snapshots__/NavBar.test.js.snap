// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <nav
    class="nav-container navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Petfinder
    </a>
    <div
      class="nav-links"
    >
      <a
        class="nav-link"
        href="/pets"
      >
        Pets
      </a>
      <a
        class="nav-link"
        href="/user/favorites"
      >
        Favorites
      </a>
      <a
        class="nav-link"
        href="/user/profile"
      >
        Profile
      </a>
      <a
        class="nav-link"
        href="/"
      >
        Logout 
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="nav-container navbar navbar-expand-md"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Petfinder
    </a>
    <div
      class="nav-links"
    >
      <a
        class="nav-link"
        href="/login"
      >
        Login
      </a>
      <a
        class="nav-link"
        href="/signup"
      >
        Signup
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
